finetune/target_models/classifier.py

```diff
@@ -209,3 +234,156 @@ class NoisyClassifier(Classifier, BaseModel):
 
     def _get_input_pipeline(self):
         return NoisyClassificationPipeline(self.config)
+
+
+class LongDocClassifier(Classifier, BaseModel):
+    """
+    Classifies a single document into 1 of N categories.
+    :param config: A :py:class:`finetune.config.Settings` object or None (for default config).
+    :param \**kwargs: key-value pairs of config items to override.
+    """
+
+    def __init__(self, **kwargs):
+        # This is somewhat messy because we also get the default config in the base model init,
+        # but needed to set max_length properly based on max_chunk_length and max_num_chunks
+        default_conf = get_default_config()
+
+        # For now, the only supported base model is LongDocBERT
+        kwargs["base_model"] = LongDocBERT
+
+        # Does not make sense to chunk long sequences
+        kwargs["chunk_long_sequences"] = False
+
+        # Modify max_length based on max_num_chunks
+        # Doing this before initializing anything to avoid shape mismatches with base model
+        # Forcibly setting max_chunk_length when set to auto is a hack
+        max_length = kwargs.get("max_length", default_conf["max_length"])
```

Comment:
These settings should be defaults of the base_model not the target model?

Reply:
`get_optimal_params` I think will work if you need to do this kind of inference. - I still think that starting with a user set max_length and chunk size and working back to num chunks is the cleaner way to do it. How it is here overriding max_length will be confusing for anyone reading back the configs and it is inconsistent with other models.